### PR TITLE
Yaku calculator

### PR DESCRIPTION
**Pull request by rasca0027**
# Description

Changes proposed in this pull request:

Implementation for `YakuCalculator`.

## Type of change

Please delete options that are not relevant.

- [x] New feature

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the tests and make sure they pass
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: mahjong/yaku_types.py [hunk 1]
```diff
@@ -242,7 +265,7 @@ def ippatsu(self):  # 一発
         1 han
         http://arcturus.su/wiki/Ippatsu
         """
-        return NotImplemented
+        return False
```
Comment (rasca0027) [inline mahjong/yaku_types.py:268]: 把NotImplemented改成False因為NotImplemented 會是True

File: mahjong/yaku_calculator.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        yaku_list = [yaku for yaku, han in yakus]
+        for yaku_name in yaku_list:
+            if yaku_name in YAKU_EXCLUDE_TABLE:
+                excluded_yakus = YAKU_EXCLUDE_TABLE[yaku_name]
+                yakus = list(filter(
+                    lambda x: x[0] not in excluded_yakus, yakus))
+        return yakus
+
+    def calculate_fu(self):
+        pass
```
Comment (rasca0027) [inline mahjong/yaku_calculator.py:73]: 不確定這裡怎弄誒

File: mahjong/yaku_types.py [hunk 3]
```diff
@@ -543,6 +583,21 @@ def tanyao(self):  # 断么九
...
 class Yakuhai(TeYaku):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.use_chain = False
+
+    def get_all_evals(self) -> List[Callable]:
+        return [
+            self.daisangen,
+            self.tsuuiisou,
```
Comment (rasca0027) [inline mahjong/yaku_types.py:594]: 字一色拉出來！

File: mahjong/yaku_calculator.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        """
+        possible_yakus: List[Tuple[str, int]] = []
+        for evaluation in self.evaluations:
+            all_evals = evaluation.get_all_evals()
+            use_chain = evaluation.use_chain
+
+            for current_eval in all_evals:
+                if current_eval():
+                    possible_yakus.append((
+                        evaluation.total_yaku[-1], evaluation.total_han[-1]))
```
Comment (rasca0027) [inline mahjong/yaku_calculator.py:47]: 役滿分開處理

**Review by hoaaoh — APPROVED** [review]

**Review by ktc312 — APPROVED** [review]

**Pull request merged**